helpers/subdocpath/subdocpath.go

```diff
@@ -0,0 +1,98 @@
+package subdocpath
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+)
+
+type Element struct {
+	Key   string
+	Index int
+}
+
+func Parse(path string) ([]Element, error) {
```

Comment:
Does this method handle escape sequences? For example:

    {
      "x.y": "a",  // subdoc path "`x.y`" or  "x`.`y"
      "x`y": "b"   // subdoc path "x``y" or "`x``y`"
    }

I think the escaping rule is that a backtick starts an escaped region, unless it's immediately followed by another backtick, in which case the backtick pair represents a literal backtick.

Reply:
Good point.  I'll go look this up in the sub-doc spec!

Reply:
I've updated the PR to correctly handle backtick escaped paths.